### PR TITLE
MWPW-142435 -Fix for fetching the correct LCP

### PR DESCRIPTION
**Pull request by Ruchika4**
* Fix for fetching the correct LCP

Resolves: [MWPW-142435](https://jira.corp.adobe.com/browse/MWPW-142435)

**Test URLs:**
- Before: https://stage--cc--adobecom.hlx.live/drafts/ruchika/batch2-performance/logo-design-software?martech=off
- After: https://lcp--cc--adobecom.hlx.live/drafts/ruchika/batch2-performance/logo-design-software?martech=off

Note: this is interim fix to make sure we load the correct LCP image for performance improvement. loadLCPImage function needs more optimizations

Need to work with team to discuss all the checks we have added for loadLCPImage and optimize more

### CODE REVIEW

File: creativecloud/scripts/utils.js [hunk 1]
```diff
@@ -59,21 +59,22 @@ function getDecorateAreaFn() {
...
   async function loadLCPImage(area = document, { fragmentLink = null } = {}) {
-    const firstBlock = area.querySelector('body > main > div > div');
+    const marquee = area.querySelector('.marquee');
+    const interactiveMarquee = area.querySelector('.interactive-marquee');
     let fgDivs = null;
     switch (true) {
-      case firstBlock?.classList.contains('changebg'): {
-        firstBlock.querySelector(':scope > div:nth-child(1)').querySelectorAll('img').forEach(eagerLoad);
+      case marquee != null:
+        eagerLoad(marquee.querySelector('img'));
```
Comment (honstar) [inline creativecloud/scripts/utils.js:67]: This assumes that the marquee is using an actual image. I know there have been usages of CSS-gradient based marquees that were later changed back to image-based ones. Maybe we should add a check here for the presence of an `img` to account for that?

File: creativecloud/scripts/utils.js [hunk 2]
```diff
@@ -59,21 +59,22 @@ function getDecorateAreaFn() {
   };
 
   async function loadLCPImage(area = document, { fragmentLink = null } = {}) {
-    const firstBlock = area.querySelector('body > main > div > div');
```
Comment (salonijain3) [inline creativecloud/scripts/utils.js:62]: @Ruchika4 In Milo also, the first div is getting selected :   `const firstDiv = document.querySelector('body > main > div:nth-child(1) > div');`

Does this change need to go in Milo?

File: creativecloud/scripts/utils.js (hunk 1 above)
Comment (honstar) [reply creativecloud/scripts/utils.js:67]: Never mind, the `eagerLoad` function already checks for null so we're good here.

File: creativecloud/scripts/utils.js [hunk 3]
```diff
@@ -59,21 +59,22 @@ function getDecorateAreaFn() {
...
         import(`${getConfig().codeRoot}/deps/interactive-marquee-changebg/changeBgMarquee.js`);
         break;
       }
-      case firstBlock?.classList.contains('marquee'):
-        firstBlock.querySelectorAll('img').forEach(eagerLoad);
-        break;
-      case firstBlock?.classList.contains('interactive-marquee'):
-        firstBlock.querySelector(':scope > div:nth-child(1)').querySelectorAll('img').forEach(eagerLoad);
-        fgDivs = firstBlock.querySelector(':scope > div:nth-child(2)').querySelectorAll('div:not(:first-child)');
+      case interactiveMarquee != null:
```
[the remark this answers is not in the document]
Comment (Ruchika4) [reply creativecloud/scripts/utils.js]: This has been updated

**Comment by honstar** [comment]
@Ruchika4 , what's the status on this? LCP is pretty important so merging this in would be great.

**Comment by Ruchika4** [comment]
> @Ruchika4 , what's the status on this? LCP is pretty important so merging this in would be great.

Hi @honstar , I am waiting for approvals. However recently team has suggested authors to move breadcrumbs down and not be the first block for other performance reasons. We did this code update initially because the code was looking for first div for LCP which was breadcrumbs. cc @salonijain3 @aishwaryamathuria

**Comment by salonijain3** [comment]
Closing this @Ruchika4 as it seems stale. Can you reopen this if this is still needed?

**Pull request closed without merging**